.gitignore

```diff
@@ -0,0 +1 @@
+dist
```

Comment:
I added a few more from there, and one extra line for emacs users, but most of the files listed there are either put inside `dist` by cabal automatically, or ghc can be instructed to put them there. I didn't include `cabal-dev` because cabal-dev is obsoleted by cabal sandbox, which is part of cabal-install.